modules/free/src/main/scala/doobie/WeakAsync.scala

```diff
@@ -7,13 +7,14 @@ package doobie
 import cats.~>
 import cats.implicits._
 import cats.effect.kernel.{ Async, Poll, Resource, Sync }
-import cats.effect.implicits._
 import cats.effect.std.Dispatcher
 import scala.concurrent.Future
 import scala.concurrent.duration.FiniteDuration
 
 trait WeakAsync[F[_]] extends Sync[F] {
   def fromFuture[A](fut: F[Future[A]]): F[A]
+  def fromFutureCancelable[A](fut: F[(Future[A], F[Unit])]): F[A] =
+    fromFuture(map(fut)(_._1)) // default impl for backwards compat; implementors should override
```

Comment:
Happy to remove this default impl, as we are already breaking compatibility, so it's better to leave this out to ensure current/future implementations are handling cancellations :)